### PR TITLE
ISIS trap support for SAI

### PR DESCRIPTION
**Pull request by VenkatCisco**
Signed-off-by: Venkat Garigipati <venkatg@cisco.com>

Extends SAI-v1.8 capabilities to support new ISIS header SAI_HOSTIF_TRAP_TYPE_ISIS.

### CODE REVIEW

**Comment by anshuv-mfst** [comment]
@rlhui - Could you please advise next steps on this PR. Maybe schedule in SAI community meeting.

**Review by kcudnik — APPROVED** [review]

File: inc/saihostif.h [hunk 1]
```diff
@@ -370,6 +370,12 @@ typedef enum _sai_hostif_trap_type_t
      */
     SAI_HOSTIF_TRAP_TYPE_BFDV6 = 0x00004006,
 
+    /**
+     * @brief Intermediate System-to-Intermediate System (IS-IS) protocol
+     * default packet action is trap
+     */
+    SAI_HOSTIF_TRAP_TYPE_ISIS = 0x00004007,
```
Comment (abdosi) [inline inc/saihostif.h:377]: please add what is match for ISIS in comments.

Comment (VenkatCisco) [reply inc/saihostif.h:379]: thanks. Added details in comments. Please review.

File: inc/saihostif.h [hunk 2]
```diff
@@ -370,6 +370,14 @@ typedef enum _sai_hostif_trap_type_t
...
     SAI_HOSTIF_TRAP_TYPE_BFDV6 = 0x00004006,
 
+    /**
+     * @brief Intermediate System-to-Intermediate System (IS-IS) protocol
+     * traffic (L1 IS Device: 01:80:c2:00:00:14, L2 IS Device: 01:80:c2:00:00:15,
+     * All IS device: 09:00:2b:00:00:05, All ES device: 09:00:2b:00:00:04).
+     * Default packet action is trap
+     */
+    SAI_HOSTIF_TRAP_TYPE_ISIS = 0x00004007,
+
```
Comment (itaibaz) [inline inc/saihostif.h:380]: ISIS is layer 2 AFAIK
And the trap is added here in the IP2ME section of traps
Please add in switch traps before SAI_HOSTIF_TRAP_TYPE_SWITCH_CUSTOM_RANGE_BASE
In addition, default action of trap, IMO, isn't backward compatible. Should be default forward

**Review by itaibaz — APPROVED** [review]

File: inc/saihostif.h [hunk 3]
```diff
@@ -322,6 +322,14 @@ typedef enum _sai_hostif_trap_type_t
     /** Default packet action is forward */
     SAI_HOSTIF_TRAP_TYPE_IPV6_NEIGHBOR_ADVERTISEMENT = 0x00002013,
 
+    /**
+     * @brief Intermediate System-to-Intermediate System (IS-IS) protocol
+     * traffic: L1: 01:80:c2:00:00:15, L2: 01:80:c2:00:00:15
```
Comment (itaibaz) [inline inc/saihostif.h:327]: L1 mac should probably be :14 and not :15

**Review by itaibaz — APPROVED** [review]

**Comment by rlhui** [comment]
could the DCO be fixed?

**Comment by VenkatCisco** [comment]
> could the DCO be fixed?

Sure, Done.

**Comment by rlhui** [comment]
> > could the DCO be fixed?
> 
> Sure, Done.

Hi, it still shows DCO not yet passed?  Thanks.

File: inc/saihostif.h [hunk 4]
```diff
@@ -322,6 +322,16 @@ typedef enum _sai_hostif_trap_type_t
     /** Default packet action is forward */
     SAI_HOSTIF_TRAP_TYPE_IPV6_NEIGHBOR_ADVERTISEMENT = 0x00002013,
 
+    /**
+     * @brief Intermediate System-to-Intermediate System (IS-IS) protocol
+     * traffic:
+     * L1 IS: 01:80:c2:00:00:14
+     * L2 IS: 01:80:c2:00:00:15
+     * All IS: 09:00:2b:00:00:05
```
Comment (kcudnik) [inline inc/saihostif.h:330]: Are those mac addresses part of this protocol standard ? also please make this more like this:

```
 * @brief Intermediate System-to-Intermediate System (IS-IS) protocol.
 *
 * Traffic: (please explain here what this traffic actually means, rather just 1 word)
 * L1 IS: 01:80:c2:00:00:14
 * L2 IS: 01:80:c2:00:00:15
 * All IS: 09:00:2b:00:00:05
 * 
 * Default packet action is forward.
```

Comment (VenkatCisco) [reply inc/saihostif.h]: Done, Can you please review.

Comment (VenkatCisco) [reply inc/saihostif.h]: > > > could the DCO be fixed?
> > 
> > 
> > Sure, Done.
> 
> Hi, it still shows DCO not yet passed? Thanks.



> > > could the DCO be fixed?
> > 
> > 
> > Sure, Done.
> 
> Hi, it still shows DCO not yet passed? Thanks.

Done. Please review.

Comment (kcudnik) [reply inc/saihostif.h]: can you add spacing empty line between brief and default packet as proposed ? please also start each sentence with capital letters like Traffic word.

Comment (VenkatCisco) [reply inc/saihostif.h]: Done. Thank you. Please review.

**Review by kcudnik — APPROVED** [review]

**Comment by kcudnik** [comment]
please fix errors https://dev.azure.com/mssonic/build/_build/results?buildId=22167&view=logs&j=83516c17-6666-5250-abde-63983ce72a49&t=00be4b52-4a63-5865-8e02-c61723ad0692

**Pull request merged**